From 4d38167330910ddb15b1add5b5cef835677a29fd Mon Sep 17 00:00:00 2001
From: Peter Zijlstra <peterz@infradead.org>
Date: Fri, 15 Oct 2021 12:05:59 +0200
Subject: [PATCH] futex: Fix PREEMPT_RT build
Git-commit: 4d38167330910ddb15b1add5b5cef835677a29fd
Patch-mainline: v5.16-rc1
References: bsc#1190137 bsc#1189998

Mike reported that rcuwait went walk-about and is causing failures on
the PREEMPT_RT builds, restore it.

Reported-by: Mike Galbraith <efault@gmx.de>
Signed-off-by: Peter Zijlstra (Intel) <peterz@infradead.org>
Signed-off-by: Davidlohr Bueso <dbueso@suse.de> 

---
 kernel/futex/futex.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/kernel/futex/futex.h b/kernel/futex/futex.h
index 948fcf317681..040ae4277cb0 100644
--- a/kernel/futex/futex.h
+++ b/kernel/futex/futex.h
@@ -5,6 +5,10 @@
 #include <linux/futex.h>
 #include <linux/sched/wake_q.h>
 
+#ifdef CONFIG_PREEMPT_RT
+#include <linux/rcuwait.h>
+#endif
+
 #include <asm/futex.h>
 
 /*
-- 
2.26.2

